include/seqan/vcf_io/read_vcf.h

```diff
@@ -64,21 +64,43 @@ typedef Tag<Vcf_> Vcf;
 // Function readRecord()                                            [VcfHeader]
 // ----------------------------------------------------------------------------
 
+template <typename TNameStore, typename TNameStoreCache, typename TStorageSpec, typename TString>
 inline void
-_parseVcfContig(CharString & chromName, CharString const & headerValue)
+_readVcfContig(VcfIOContext<TNameStore, TNameStoreCache, TStorageSpec> & context, TString const & headerValue)
 {
-    if (length(headerValue) < 3u)
-        return;
+    typedef OrFunctor<EqualsChar<','>, EqualsChar<'>'> >            IsCommaOrGt;
+    typedef typename DirectionIterator<TString const, Input>::Type  TIter;
 
-    CharString tmp = infix(headerValue, 1, length(headerValue) - 2);
-    StringSet<CharString> tmp2;
-    strSplit(tmp2, tmp, EqualsChar<','>());
-    for (unsigned i = 0; i < length(tmp2); ++i)
-    {
-        if (!startsWith(tmp2[i], "ID="))
-            continue;
-        chromName = suffix(tmp2[i], 3);
-    }
+    TIter headerIter = directionIterator(headerValue, Input());
+    CharString &buffer = context.buffer;
+
+    // Seek ID key.
+    clear(buffer);
+    skipOne(headerIter);
+    readUntil(buffer, headerIter, EqualsChar<'='>());
+    if (buffer != "ID")
+        SEQAN_THROW(ParseError("Contig ID key not found in header."));
+    skipOne(headerIter);
+
+    // Read ID value.
+    clear(buffer);
+    readUntil(buffer, headerIter, IsCommaOrGt());
+    if (empty(buffer))
+        SEQAN_THROW(ParseError("Contig ID key is empty."));
+    appendName(contigNamesCache(context), buffer);
+
+    // Seek length key.
```

Comment:
Could you add a note to remove comments, when this is reflected by the IOContext.